src/bacnet/basic/object/iv.c

```diff
@@ -233,22 +233,65 @@ bool Integer_Value_Present_Value_Set(
 bool Integer_Value_Object_Name(
     uint32_t object_instance, BACNET_CHARACTER_STRING *object_name)
 {
+    static char text[32] = ""; /* okay for single thread */
     bool status = false;
+    struct integer_object *pObject;
 
-    struct integer_object *pObject = Integer_Value_Object(object_instance);
-
-    if (!object_name) {
-        return false;
+    pObject = Integer_Value_Object(object_instance);
+    if (pObject) {
+        if (pObject->Object_Name) {
+            status =
+                characterstring_init_ansi(object_name, pObject->Object_Name);
+        } else {
+            snprintf(
+                text, sizeof(text), "INTEGER-VALUE-%lu",
+                (unsigned long)object_instance);
+            status = characterstring_init_ansi(object_name, text);
+        }
     }
 
+    return status;
+}
+
+/**
+ * @brief For a given object instance-number, sets the object-name
+ * @param  object_instance - object-instance number of the object
+ * @param  new_name - holds the object-name to be set
+ * @return  true if object-name was set
+ */
+bool Integer_Value_Name_Set(uint32_t object_instance,
+    char *new_name)
```

Comment:
Should most of the 'const' changes be done in [#714] except for the new functions?

Reply:
Yeah definetly. This is new function right?

Reply:
And I meant: search other places where you have added new Name_Set function.

Reply:
I think it will be better to not change to 'const' until after [#714] since I made a common testing function that does not use 'const'
```
/* function API pattern for testing ASCII name get/set */
typedef bool (*object_name_ascii_set_function) (uint32_t object_instance,
    char *new_name);
typedef const char * (*object_name_ascii_function) (
    uint32_t object_instance);
```
Is [#714] ready for merging?

Reply:
Sure. I just fear that if you have something big coming up then you get way too many conflicts. If you are ok resolving those then go ahead and merge 714. I just pushed one trailing whitespace fixes for it.

Also https://github.com/bacnet-stack/bacnet-stack/pull/744 can be merged if that is ok. I then push https://github.com/bacnet-stack/bacnet-stack/pull/753 again.

Reply:
And it is also totally ok to merge this without changing const. We anyway have lot of things which needs to be done before we get everything const correct.